Deprecate torch.lu

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73878
* #73877
* #74046
* #73806
* __->__ #73804
* #73803
* #72935

**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

lezcano committed:
Deprecate torch.lu

**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

IvanYashchuk commented on aten/src/ATen/native/BatchLinearAlgebra.cpp:
```diff
@@ -1795,6 +1795,17 @@ std::tuple<Tensor, Tensor> linalg_lu_factor(const Tensor& A, bool pivot) {
+   TORCH_WARN_ONCE(
+    "torch.lu is deprecated in favor of torch.linalg.lu_factor / torch.linalg.lu_factor_ex and will be ",
+    "removed in a future PyTorch release.\n",
+    "LU, pivots, info = torch.lu(A, compute_pivots)\n",
```
```suggestion
    "LU, pivots = torch.lu(A, compute_pivots)\n",
```

IvanYashchuk commented on aten/src/ATen/native/BatchLinearAlgebra.cpp:
```diff
@@ -1795,6 +1795,17 @@ std::tuple<Tensor, Tensor> linalg_lu_factor(const Tensor& A, bool pivot) {
+    "and\n",
+    "LU, pivots, info = torch.lu(A, compute_pivots, get_infos=True)\n",
+    "should be replaced with\n",
+    "LU, pivots, info = torch.linalg.lu_factor_ex(A, compute_pivots, false)"
```
`check_errors` is False by default.
```suggestion
    "LU, pivots, info = torch.linalg.lu_factor_ex(A, compute_pivots)"
```

IvanYashchuk commented on torch/functional.py:
```diff
@@ -1673,6 +1673,21 @@ def _lu_impl(A, pivot=True, get_infos=False, out=None):
+        :func:`torch.lu` is deprecated in favor of :func:`torch.linalg.lu_factor`
+        and :func:`torch.linalg.lu_factor_ex`. :func:`torch.lu` will be removed in a
+        future PyTorch release.
+        ``LU, pivots, info = torch.lu(A, compute_pivots)`` should be replaced with
```
```suggestion
        ``LU, pivots = torch.lu(A, compute_pivots)`` should be replaced with
```

IvanYashchuk commented on torch/functional.py:
```diff
@@ -1673,6 +1673,21 @@ def _lu_impl(A, pivot=True, get_infos=False, out=None):
+        ``LU, pivots, info = torch.lu(A, compute_pivots, get_infos=True)`` should be replaced with
+
+        .. code:: python
+            LU, pivots, info = torch.linalg.lu_factor_ex(A, compute_pivots, false)
```
```suggestion
            LU, pivots, info = torch.linalg.lu_factor_ex(A, compute_pivots)
```

IvanYashchuk commented:
There are a few problems with the suggested code in the warning message.

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

Lezcano commented:
Fixed, thanks for spotting these!

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

lezcano committed:
Update on "Deprecate torch.lu"


**BC-breaking note**:

This PR deprecates `torch.lu` in favor of `torch.linalg.lu_factor`.
A upgrade guide is added to the documentation for `torch.lu`.

Note this PR DOES NOT remove `torch.lu`.

[ghstack-poisoned]

mruberry commented on aten/src/ATen/native/BatchLinearAlgebra.cpp:
```diff
@@ -2256,6 +2256,17 @@ std::tuple<Tensor, Tensor> linalg_lu_factor(const Tensor& A, bool pivot) {
 
 // TODO Deprecate this function in favour of linalg_lu_factor_ex
 std::tuple<Tensor, Tensor, Tensor> _lu_with_info(const Tensor& self, bool compute_pivots, bool) {
+   TORCH_WARN_ONCE(
```
Nice warn once use

mruberry reviewed (approved):
Nice deprecation - stamped!

Lezcano commented:
@pytorchmergebot merge this please

malfet commented:
@pytorchbot revert this, as dependent change of https://github.com/pytorch/pytorch/pull/72935 that breaks internal build

Lezcano commented:
Relanding in https://github.com/pytorch/pytorch/pull/77636
